CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/blaze/BlazeActions.java

```diff
@@ -189,19 +469,130 @@ private void setBlazeProperties(final float size, final float opacity) {
+        }
+
+        final List<ConstellationColor> colors = getPresetCustomColors();
+
+        if (colors != null) {
+            int panelCounter = 0;
+            for (final JPanel panel : colorPanels) {
+                if (panelCounter < colors.size()) {
+                    if (colors.get(panelCounter) != null) {
+                        panel.setBackground(colors.get(panelCounter).getJavaColor());
+                    }
+                }
+                panelCounter++;
+            }
+        }
+        setBlazeItems(BlazeActions.getPresetCustomColors(), this.presetCustomBlazeItems);
+    }
+
+    /**
+     * Set the items for the JMenuItem blazeItems with the colors passed in
+     * throught the colors parameter.
+     *
+     * @param colors - the colors to set in the menu
+     * @param blazeItems - the menu item to set the color of
+     */
+    private void setBlazeItems(final List<ConstellationColor> colors, final JMenuItem[] blazeItems) {
+        Collections.reverse(colors);
+        int idx = 0;
+        for (final ConstellationColor color : colors) {
+            if (color != null) {
+
+                final Color javaColor = color.getJavaColor();
+                String colorName = "#" + String.format("%02x", javaColor.getRed())
+                        + String.format("%02x", javaColor.getGreen())
+                        + String.format("%02x", javaColor.getBlue());
+
+                if (color.getName() != null) {
+                    colorName = color.getName();
+                }
+
+                blazeItems[BlazeUtilities.MAXIMUM_CUSTOM_BLAZE_COLORS - idx - 1].setText(colorName);
+                blazeItems[BlazeUtilities.MAXIMUM_CUSTOM_BLAZE_COLORS - idx - 1].setActionCommand(ADD_PRESET_BLAZE_ACTION + colorName);
+                blazeItems[BlazeUtilities.MAXIMUM_CUSTOM_BLAZE_COLORS - idx - 1].setVisible(true);
+                blazeItems[BlazeUtilities.MAXIMUM_CUSTOM_BLAZE_COLORS - idx - 1].setIcon(generateCustomImage(javaColor));
+                idx++;
+            }
+
+        }
+    }
+
+    /**
+     * Generate the color of a menu item by specifying the color and return a
+     * suitable ImageIcon
+     *
+     * @param javaColor - the color to paint
+     * @return the painted ImageIcon
+     */
+    private ImageIcon generateCustomImage(final Color javaColor) {
+        final BufferedImage customImage = this.copyImageBuffer((BufferedImage) ImageUtilities.loadImage(ADD_RECENT_BLAZE_ICON, false));
+        for (int x = 0; x < customImage.getWidth(); x++) {
+            for (int y = 0; y < customImage.getHeight(); y++) {
+                if (customImage.getRGB(x, y) == BLACK_COLOR) {
+                    customImage.setRGB(x, y, javaColor.getRGB());
+                }
+            }
+        }
+        return new ImageIcon(customImage);
+    }
+
+    private void colorPanel1MouseClicked(MouseEvent evt) {
```

Comment:
I feel like these functions aren't necessary given the parameter isn't used and it's only a one line function. Would probably be better to just directly call the function